### DISCUSSION TITLE
best practics for using together with react-table

### DISCUSSION THREAD

**Discussion by chiptus** [TanStack/query#3321] [Q&A]
for some reason when I use `useQuery` and `useTable` in the same component I get infinite renders and the app crashes. for now I put `useQuery` in a container component that then renders the table only once data is available (behined `if (query.data) ...`). This works, but I need to create a wrapping component (and name it! :) )

I'll try to get a codesandbox as an example. but I was wondering how to community works with both libraries and what are the best practices for this?

**edit**: I tried searching both libraries issues and discussions but didn't find a similar question

**Comment by TkDodo** [comment]
I can take a look at the codesandbox example if you provide it. I think it would also make for a neat example :)

**Comment by chiptus** [comment]
ok, my example was very complex because of the use of useEffect to setGlobalFilter and hideColumns on load. this simple example works: https://codesandbox.io/s/react-table-query-75jkb3

after seeing this example I succeeded refactoring my table to use the query inside
